### PR TITLE
Add Trino 384 release notes

### LINKED ISSUE #12644: Release notes for 384

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 384 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Docs only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 384 with sections for general, security, connectors and other topics as included in the release.

## Related issues, pull requests, and links

* Fixes #12644 

## Documentation

( ) No documentation is needed.
(✅) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅) No release notes entries required.
( ) Release notes entries required with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 2 June 2022 ✅ 
81afa439420d436b0a68af1d94fb98c91b23d10e @sopel39 #12551 ✅ rn ✅ docs
1ee7f9377020400ff4808c684c53a90fb318aa93 @sopel39 #12551 ✅ rn ✅ docs
639121e1bd9bd43c44435f5ccbf02efcc4842e23 @sopel39 #12551 ✅ rn ✅ docs
c1e7e1bfe02792090e821380ea5e670c7087b30c @sopel39 #12551 ✅ rn ✅ docs
38ce359ff1999d9c23805d760183ebf963d44895 @sopel39 #12551 ✅ rn ✅ docs
4203b904f381a151ec7afb44cd6680595d2ca478 @findepi #12651 ✅ rn ✅ docs

## 3 June 2022 ✅ 
8c138334960cb52a5bd2852daeb55a17ec45f1a1 @sopel39 #12670 ✅ rn ✅ docs
5d65784c3b5fa79c6448c5c874e4cea0fa371cd3 @hashhar #12650 ✅ rn ✅ docs
3c7041911797693eaa524ebfbdade4b90becccbd @findepi #12325  ✅ rn ✅ docs
14d00cf3dd55a14be7eb90450395e35976be69d3 @findepi #12325 ✅ rn ✅ docs
159a05cb7074dd20f9958620ddf84c912c7eabcd @findepi #12325 ✅ rn ✅ docs
2a7e4977b6049df960fbcf34f0e42c207cbf68c8 @findepi #12325 ✅ rn ✅ docs
65115894f53e4f836947718558b222f6fd0d1abc @findepi #12325 ✅ rn ✅ docs
117b2e7d2a2e046e8ae66e9b6e58a487d216276c @findepi #12325 ✅ rn ✅ docs
c1ea26900ea10a5162dbbeda49d197b048a2ec53 @findepi #12595 ✅ rn ✅ docs
4aacac24cb2811970abaa5acfa47427206568562 @findepi #12595 ✅ rn ✅ docs
c203ec8eed7315b7576a59280e3669e3124983ed @martint #12681 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-384.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Release 384 (3 June 2022)
+
+## General
+
+* Fix queries failing due to worker data exchange optimization introduced in
+  Trino 383. ({issue}`12670`)
+* Fix potential failure or incorrect result of `json_query` function. ({issue}`12681`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-384.md:7]: of the

File: docs/src/main/sphinx/release/release-384.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Druid connector
+
+* Add `query` table function for full query pass-through to the connector. ({issue}`1839`,`9163`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-384.md:11]: These should link only to issue 9163

Comment (mosabua) [inline docs/src/main/sphinx/release/release-384.md:11]: does that work in terms of output

{issue}`1839`,`9163`

in the past I always did

{issue}`1839`, {issue}`9163`

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-384.md (hunk 1 above)
Comment (martint) [inline docs/src/main/sphinx/release/release-384.md:7]: Maybe: 

> Fix potential query failure or incorrect results for `json_query` function.

Also, link to the function docs.

File: docs/src/main/sphinx/release/release-384.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Release 384 (3 June 2022)
+
+## General
+
+* Fix queries failing due to worker data exchange optimization introduced in
```
Comment (martint) [inline docs/src/main/sphinx/release/release-384.md:5]: I would just phrase it as:

> Fix potential query failure due to a regression introduced in version 383.

The fact that it's due to an optimization in exchanges is irrelevant to users that just want to know if the issue they are encountering is fixed.

We may also want to add a note in the 383 release notes saying:

> This version has a regression that may cause queries to fail.

File: docs/src/main/sphinx/release/release-383.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # Release 383 (1 June 2022)
 
+Note: This version has a regression that may cause queries to fail.
```
Comment (martint) [inline docs/src/main/sphinx/release/release-383.md:3]: Format using a "warning" block. See the 349 release notes for an example.

**Comment by martint** [comment]
Merged as c991e76a5b2de76e26aaedf0fff1e5452e781920

**Pull request closed without merging**